std-bits/base/src/main/java/org/enso/base/DryRunFileManager.java

```diff
@@ -0,0 +1,40 @@
+package org.enso.base;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+
+public class DryRunFileManager {
+  static final Map<String, String> files = new HashMap<>();
+
+  /**
+   * Creates a temporary file for the given name.
+   *
+   * @param name the path to the file to make a temporary file of.
+   * @return the path to the temporary file.
+   */
+  public static String getTemporaryFile(String name) {
+    return files.computeIfAbsent(
+        name,
+        k -> {
+          if (files.containsValue(k)) {
+            // Existing temporary file so return this.
+            return k;
+          }
+
+          var filename = new File(k).getName();
+          var lastDot = filename.lastIndexOf('.');
+          var prefix = lastDot == -1 ? filename : filename.substring(0, lastDot);
+          var extension = lastDot == -1 ? "" : filename.substring(lastDot);
+
+          try {
+            var temp = File.createTempFile(prefix, extension);
```

Comment:
I'm wondering if it's worth to prepend to all prefixes sth like `enso-dry-run-` to make the files more easily identifiable.

Feel free to ignore this though.